scripts/Muon/GUI/ElementalAnalysis/Plotting/plotting_view.py

```diff
@@ -0,0 +1,219 @@
+from collections import OrderedDict
+
+from six import iteritems
+
+from mantid import plots
+
+from PyQt4 import QtGui
+
+from matplotlib.figure import Figure
+from matplotlib import gridspec
+from matplotlib.backends.backend_qt4agg import FigureCanvasQTAgg as FigureCanvas
+# pyplot should not be imported:
+# https://stackoverflow.com/posts/comments/26295260
+
+from Muon.GUI.ElementalAnalysis.Plotting import plotting_utils as putils
+from Muon.GUI.ElementalAnalysis.Plotting.AxisChanger.axis_changer_presenter import AxisChangerPresenter
+from Muon.GUI.ElementalAnalysis.Plotting.AxisChanger.axis_changer_view import AxisChangerView
+
+
+class PlotView(QtGui.QWidget):
+    def __init__(self):
+        super(PlotView, self).__init__()
+        self.plots = OrderedDict({})
+        self.workspaces = {}
+        self.plot_additions = {}
+        self.current_grid = None
+        self.gridspecs = {
+            1: gridspec.GridSpec(1, 1),
+            2: gridspec.GridSpec(1, 2),
+            3: gridspec.GridSpec(3, 1),
+            4: gridspec.GridSpec(2, 2)
+        }
+        self.figure = Figure()
+        self.figure.set_facecolor("none")
+        self.canvas = FigureCanvas(self.figure)
+
+        self.plot_selector = QtGui.QComboBox()
+        self.plot_selector.currentIndexChanged[str].connect(self._set_bounds)
+
+        button_layout = QtGui.QHBoxLayout()
+        self.x_axis_changer = AxisChangerPresenter(AxisChangerView("X"))
+        self.x_axis_changer.on_bounds_changed(self._update_x_axis)
+
+        self.y_axis_changer = AxisChangerPresenter(AxisChangerView("Y"))
+        self.y_axis_changer.on_bounds_changed(self._update_y_axis)
+
+        self.errors = QtGui.QCheckBox("Errors")
+        self.errors.stateChanged.connect(self._errors_changed)
+
+        button_layout.addWidget(self.plot_selector)
+        button_layout.addWidget(self.x_axis_changer.view)
+        button_layout.addWidget(self.y_axis_changer.view)
+        button_layout.addWidget(self.errors)
+
+        grid = QtGui.QGridLayout()
+        grid.addWidget(self.canvas, 0, 0)
+        grid.addLayout(button_layout, 1, 0)
+        self.setLayout(grid)
+
+    def _redo_layout(func):
+        """
+        Simple decorator to call tight_layout() on plots, and to redraw the canvas.
+        """
+
+        def wraps(self, *args, **kwargs):
+            func(self, *args, **kwargs)
+            if len(self.plots):
+                self.figure.tight_layout()
+            self.canvas.draw()
+        return wraps
+
+    def _save_addition(func):
```

Comment:
where is the code for this and should the ``wraps`` method be inside this?

Reply:
The `wraps` method should be inside - it allows functions to be [decorated](https://www.python.org/dev/peps/pep-0318/) with `_redo_layout` (`@_redo_layout`).

Reply:
can we have this in a comment (including the link)?